### PR TITLE
Changes exam report status and score columns

### PR DESCRIPTION
**Pull request by christianmemije**
## Summary

* Changes exam report status and score columns
* Addresses #1431 

## Before

image

image

## Now
image

image

### CODE REVIEW

**Comment by khangmach** [comment]
@christianmemije perfect

File: kolibri/plugins/coach/assets/src/views/exam-report-page/index.vue [hunk 1]
```diff
@@ -37,19 +37,19 @@
...
+                {{ $tr('remaining', { num: (exam.question_count - examTaker.progress) }) }}
               </span>
               <span v-else>
                 {{ $tr('notstarted') }}
               </span>
             </td>
 
             <td class="table-data">
-              <span v-if="examTaker.progress === undefined">&mdash;</span>
+              <span v-if="!examTaker.score">&mdash;</span>
```
Comment (indirectlylit) [inline kolibri/plugins/coach/assets/src/views/exam-report-page/index.vue]: does this distinguish between a score of 0% and an undefined score?

Comment (rtibbles) [reply kolibri/plugins/coach/assets/src/views/exam-report-page/index.vue]: It will not, this should be reverted.

**Comment by christianmemije** [comment]
@rtibbles Should be checking for `null` not `undefined`.

**Checking for `undefined` results in:** 

image

**Comment by rtibbles** [comment]
This is intended behaviour - once you have started an exam, it shows a score (even if you have not answered any questions yet).

**Comment by christianmemije** [comment]
@khangmach

**Comment by khangmach** [comment]
@christianmemije @rtibbles 
A score seems like something that naturally pops up only after completion. Otherwise, how can you be scored on something you haven't submitted? Even if we calculate score in real time, if answers are changed by the learner while test-taking, we'll always have this fluctuating number that is mixed with scores of those who have completed the exam. 

I would opt for either a fixed 0% for in-progress learners or not display a score at all. That way it's easier to visually distinguish those who have completed vs in progress

**Comment by rtibbles** [comment]
Unless the teacher closed the exam part way through a user taking it, in which case it will be forever incomplete, but needs to be scored.

**Comment by khangmach** [comment]
@richard Does the above imply that If the teacher closed the exam, Kolibri finalizes all scores? my suggestion was to just hide the in progress scores or replace with "--", and then if the exam is closed part way through, the final scores appear.

**Comment by rtibbles** [comment]
That sounds like a reasonable way to do it also - my only question is what we gain by hiding information from the teacher (is there a benefit to a 'big reveal' when the user has finished?)

**Comment by khangmach** [comment]
@rtibbles We're hiding for good reason. If we show the score before the exam completes, the score is still tentative and not finalized. The teacher probably knows this and will understand that there probably isn't a real reason for them to stare consistently at fluctuating scores, because, well, they're fluctuating and won't be set in stone until the exam ends. 

Instead, what's the real use of this exam view? Well, probably to get a quick glimpse at how many students finished, and who's still in progress. The more easily we can create a visual separation between the two, the better. It probably wouldn't help if there were percentages everywhere. If there were, teachers would be forced to check the status column instead of the percent column, which is even more reading. The purpose of the % column would be defeated.

The sacrifice overall makes for a more readable experience IMO.

**Pull request merged**